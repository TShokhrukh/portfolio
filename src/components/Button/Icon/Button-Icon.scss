.Button .Button-Icon {
  display: inline-block;
  font-size: 24px;
}

.Button-Icon_direction_start {
  margin-right: $spacer;
}

.Button-Icon_direction_end {
  margin-left: $spacer;
}
