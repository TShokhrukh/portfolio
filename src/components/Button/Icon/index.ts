export { ButtonIcon as default } from './Button-Icon'
