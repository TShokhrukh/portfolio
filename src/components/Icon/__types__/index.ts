import { IClassNameProps } from '@bem-react/core'

export interface IIconProps extends IClassNameProps {
  name: string
  id?: string
}
