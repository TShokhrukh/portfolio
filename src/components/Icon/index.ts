export * from './build'
export * from './__types__'
export { Icon as IconBase } from './Icon'
