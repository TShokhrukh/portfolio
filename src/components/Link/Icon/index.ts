export { LinkIcon as default } from './Link-Icon'
