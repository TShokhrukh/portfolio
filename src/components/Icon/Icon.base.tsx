import React, { FC } from 'react'
import { cnIcon } from './Icon.const'
import type { IIconProps } from './__types__'

import './Icon.scss'

export const IconBase: FC<IIconProps> = ({
  name,
  className,
  ...props
}) => (
  <i {...props} className={cnIcon({}, [className])}
    >{name}</i>
)
