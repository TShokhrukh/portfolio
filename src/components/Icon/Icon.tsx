import React, { FC } from 'react'
import FontAwesomeIcon from 'react-fontawesome'
import faStyles from 'font-awesome/css/font-awesome.css'
import { cnIcon } from './Icon.const'
import type { IIconProps } from './__types__'

export const Icon: FC<IIconProps> = ({
  name,
  className,
  ...props
}) => (
  <FontAwesomeIcon {...props} name={name} cssModule={faStyles} className={cnIcon({}, [className])} />
)
