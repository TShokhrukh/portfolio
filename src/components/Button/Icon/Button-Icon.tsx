import React from 'react'

import { cnButton } from '../Button.const'
import type { IButtonIconProps } from '../__types__'

import './Button-Icon.scss'

export const ButtonIcon: React.FC<IButtonIconProps> = ({
  direction,
  icon: Icon
}) => {
  return <Icon className={cnButton('Icon', { direction })} />
}
