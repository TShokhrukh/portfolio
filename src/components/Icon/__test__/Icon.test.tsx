/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { Icon } from '../Icon'

describe('Icon', () => {
  test('component template', () => {
    render(<Icon data-testid='icon' name='user' />)

    expect(screen.getByTestId('icon')).toMatchSnapshot()
  })
})
