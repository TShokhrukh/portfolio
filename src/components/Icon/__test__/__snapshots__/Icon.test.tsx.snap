// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component template 1`] = `
<span
  aria-hidden="true"
  class="fa fa-user Icon"
  data-testid="icon"
/>
`;
